library/proc_macro/src/quote.rs

```diff
@@ -105,7 +104,7 @@ pub fn quote(stream: TokenStream) -> TokenStream {
                 ))),
                 TokenTree::Ident(tt) => quote!(crate::TokenTree::Ident(crate::Ident::new(
                     (@ TokenTree::from(Literal::string(&tt.to_string()))),
-                    (@ quote_span(proc_macro_crate.clone(), tt.span())),
+                    crate::Span::def_site(),
```

Comment:
While `Span::def_site()` is unstable, so is `quote!{}`. In addition `Span::def_site()` is the closest to the behavior before this PR.